cvat/apps/lambda_manager/views.py

```diff
@@ -0,0 +1,341 @@
+                data = str(err)
+            except requests.RequestException as err:
+                status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
+                data = str(err)
+            except ValidationError as err:
+                status_code = err.code
+                data = err.message
+
+            return Response(data=data, status=status_code)
+
+        return func_wrapper
+    return wrap_response
+
+# TODO: need to protect the class. Only a registered use can get information
+# about available serverless functions.
+class FunctionViewSet(viewsets.ViewSet):
+    lookup_value_regex = '[a-zA-Z0-9_.-]+'
+    lookup_field = 'id'
+
+    @return_response()
+    def list(self, request):
+        gateway = LambdaGateway()
+        return [f.to_dict() for f in gateway.list()]
+
+    @return_response()
+    def retrieve(self, request, id):
+        gateway = LambdaGateway()
+        return gateway.get(id).to_dict()
+
+    @return_response()
+    def call(self, request, id):
+        try:
+            # Mandatory parameters
+            task = request.data['task']
+            frame = request.data['frame']
+
+            # Optional parameters
+            points = request.data.get('points')
+            quality = request.data.get('quality')
+            mapping = request.data.get('mapping')
+
+            db_task = TaskModel.objects.get(pk=task)
+        except (KeyError, ObjectDoesNotExist) as err:
+            raise ValidationError(
+                '`{}` lambda function was run '.format(id) +
+                'with wrong arguments ({})'.format(str(err)),
+                code=status.HTTP_400_BAD_REQUEST)
+
+        gateway = LambdaGateway()
+        lambda_func = gateway.get(id)
+
+        return lambda_func.invoke(db_task, frame, quality, mapping, points)
+
+# TODO: need to protect the class. Only a person with appropriate rights
+# can create the annotation request and cancel it.
+class RequestViewSet(viewsets.ViewSet):
+    @return_response()
+    def list(self, request):
+        queue = LambdaQueue()
+        return [job.to_dict() for job in queue.get_jobs()]
+
+    @return_response()
+    def create(self, request):
+        function = request.data['function']
+        threshold = request.data.get('threshold')
+        task = request.data['task']
+        quality = request.data.get("quality")
+        cleanup = request.data.get('cleanup', False)
+        mapping = request.data.get('mapping')
+
+        gateway = LambdaGateway()
+        queue = LambdaQueue()
+        lambda_func = gateway.get(function)
+        job = queue.enqueue(lambda_func, threshold, task, quality,
+            mapping, cleanup)
+
+        return job.to_dict()
+
+    @return_response()
+    def retrieve(self, request, pk):
+        queue = LambdaQueue()
+        job = queue.fetch_job(pk)
+
+        return job.to_dict()
+
+    @return_response(status.HTTP_204_NO_CONTENT)
+    def delete(self, request, pk):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5729179)